### PR TITLE
Setting up facility users to change profile

### LINKED ISSUE #917: allow profile editing to be performed by device owners
Was not implemented during #915 

Specifically, they device owner should be able to update:

* username
* password

### PR DESCRIPTION
**Pull request by DXCanas**
Fixes #917

Quick fix, seems to work fine. Do have some questions about recent changes to profile page.

### CODE REVIEW

File: kolibri/plugins/user/assets/src/state/actions.js [hunk 1]
```diff
@@ -27,7 +28,13 @@ function showRoot(store) {
 function editProfile(store, edits, session) {
   // payload needs username, fullname, and facility
   // used to save changes to API
-  const savedUserModel = FacilityUserResource.getModel(session.user_id);
+  function getUserModel() {
+    if (coreGetters.isAdmin) {
```
Comment (christianmemije) [inline kolibri/plugins/user/assets/src/state/actions.js]: Should be `coreGetters.isAdmin(store.state)`

File: kolibri/plugins/user/assets/src/views/profile-page/index.vue [hunk 2]
```diff
@@ -111,7 +111,7 @@
       },
       canEditUsername() {
         if (this.isSuperuser) {
```
Comment (christianmemije) [inline kolibri/plugins/user/assets/src/views/profile-page/index.vue]: Combine the checks for `isSuperUser` and `isAdmin`

File: kolibri/plugins/user/assets/src/views/profile-page/index.vue [hunk 3]
```diff
@@ -121,7 +121,7 @@
       },
       canEditName() {
         if (this.isSuperuser) {
```
Comment (christianmemije) [inline kolibri/plugins/user/assets/src/views/profile-page/index.vue]: Same comment as above.

**Comment by DXCanas** [comment]
Weird. I tested it and it's working fine, but it should be broken.

**Comment by DXCanas** [comment]
Updated

File: kolibri/plugins/user/assets/src/views/profile-page/index.vue (hunk 2 above)
Comment (christianmemije) [reply kolibri/plugins/user/assets/src/views/profile-page/index.vue]: Combine

File: kolibri/plugins/user/assets/src/state/actions.js [hunk 4]
```diff
@@ -27,7 +28,13 @@ function showRoot(store) {
 function editProfile(store, edits, session) {
   // payload needs username, fullname, and facility
   // used to save changes to API
-  const savedUserModel = FacilityUserResource.getModel(session.user_id);
+  function getUserModel() {
+    if (coreGetters.isAdmin(store)) {
```
Comment (christianmemije) [inline kolibri/plugins/user/assets/src/state/actions.js]: `store.state`

Comment (DXCanas) [reply kolibri/plugins/user/assets/src/state/actions.js]: :snowflake:

**Comment by DXCanas** [comment]
Updated

File: kolibri/plugins/user/assets/src/views/profile-page/index.vue [hunk 5]
```diff
@@ -110,19 +110,15 @@
...
         } else if (this.isCoach || this.isLearner) {
           return this.facilityConfig.learnerCanEditUsername;
         }
         return false;
       },
       canEditName() {
-        if (this.isSuperuser) {
-          return false;
-        } else if (this.isAdmin) {
+        if (this.isAdmin || this.isSuperuser) {
```
Comment (christianmemije) [inline kolibri/plugins/user/assets/src/views/profile-page/index.vue]: Might as well keep the order consistent `this.isSuperuser || this.isAdmin`

File: kolibri/plugins/user/assets/src/views/profile-page/index.vue [hunk 6]
```diff
@@ -110,24 +110,16 @@
         return '';
       },
       canEditUsername() {
-        if (this.isSuperuser) {
-          return true;
-        } else if (this.isAdmin) {
-          return true;
-        } else if (this.isCoach || this.isLearner) {
+        if (this.isCoach || this.isLearner) {
```
Comment (DXCanas) [inline kolibri/plugins/user/assets/src/views/profile-page/index.vue]: I like this better @christianmemije

**Pull request merged**